enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -300,57 +264,64 @@ The implementation changes needed for the proposed enhancement are documented in
-to the ACL rule(s) for the EgressFirewall rule(s) in which the DNS name is used. This will ensure that the latest IPs are always updated in
-the ``AddressSets``.
+The OVN-K master(s) will watch the `EgressFirewallDNSName` CRs. Whenever the `.status` of the CRs will be updated with new IP addresses and corresponding
+TTL information for a DNS name, OVN-K master(s) will update the `AddressSet` mapped to the DNS name. This `AddressSet` will be linked
+to the ACL rule(s) for the EgressFirewall rule(s) in which the DNS name is used. This will ensure that the latest IP addresses are always updated in
+the `AddressSets`.
 
-For wildcard DNS names, the OVN-K master will only query for the DNS names that get added to the ``.status`` of the corresponding ``EgressFirewallDNSName`` CR,
-including the wildcard DNS name if it get added to the `.status`. However, the list of the DNS names to lookup for a wildcard DNS name should also
-not become stale if a DNS name belonging its subdomain is removed. To achieve this a retry counter will be used for the DNS name
-lookups. If the lookup fails for a DNS name listed in the ``.status`` of a ``EgressFirewallDNSName`` CR for threshold number of times (say 5),
-then the DNS name will be removed from the ``.status``. However the DNS lookup will fail, if the corresponding wildcard DNS name does not have an A
-record. Otherwise, the response will be same as the IP addresses associated with the A record of the wildcard DNS name.
 
 ### Risks and Mitigations
 
-* The ``EgressFirewallDNSName`` CR will be created by OVN-K master whenever a new DNS name is used in a EgressFirewall rule. The CR will be deleted
-when the corresponding DNS name is not used in any of the EgressFirewall rules. The ``EgressFirewallDNSName`` CR should not be modified (created or deleted or
+* The `EgressFirewallDNSName` CR will be created by OVN-K Cluster manager whenever a new DNS name is used in a EgressFirewall rule. The CR will be deleted
+when the corresponding DNS name is not used in any of the EgressFirewall rules. The `EgressFirewallDNSName` CR should not be modified (created or deleted or
 updated) by an user. Doing so may lead to undesired behavior of EgressFirewall.
 
 ### Drawbacks
 
-* Whenever there's a change in the IPs or the next lookup time (TTL + last lookup time) for a DNS name, the additional step of updating the
-related ``EgressFirewallDNSName`` CRs will be executed. The `egressfirewall` plugin will wait until the IPs are updated in the corresponding
-`AddressSet` and then send the response to the DNS lookup. This will add some delay to the DNS lookup process. However, this will only
+* Whenever there's a change in the IP addresses or the next lookup time (TTL + last lookup time) for a DNS name, the additional step of updating the
+related `EgressFirewallDNSName` CRs will be executed. This will add some delay to the DNS lookup process. However, this will only
```

Comment:
Please present an estimate the delay, for one record and for a cumulative total of daily operations.   Could there possibly be a way to design this so there is never an impact if there is no egress firewall?

Reply:
> Please present an estimate the delay, for one record and for a cumulative total of daily operations.

The main delay component will be due to the call to the apiserver for updating the status of the CR, if there are any updates. Can you suggest how to provide an estimate of the delay for the same?


> Could there possibly be a way to design this so there is never an impact if there is no egress firewall?

The details of how the new plugin will perform the matching is explained here: https://github.com/openshift/enhancements/pull/1335/files#diff-b27f04c5e2673c47f922695835e746ecb2205a9f7800def32cc15c25af521758R294-R298

I have also responded in detail here: https://github.com/openshift/enhancements/pull/1335#discussion_r1156610054